sound/soc/sof/core.c

```diff
@@ -27,6 +27,22 @@ MODULE_PARM_DESC(sof_debug, "SOF core debug options (0x0 all off)");
 #define TIMEOUT_DEFAULT_IPC_MS  500
 #define TIMEOUT_DEFAULT_BOOT_MS 2000
 
+/**
+ * sof_debug_check_flag - check if a given flag(s) is set in sof_core_debug
+ * @mask: Flag or combination of flags to check
+ *
+ * Returns true if all bits set in mask is also set in sof_core_debug, otherwise
+ * false
+ */
+bool sof_debug_check_flag(int mask)
+{
+	if ((sof_core_debug & mask) == mask)
```

Comment:
this makes sense if we want to check multiple flags at once but in most cases we'd be checking if a single mask is set no? so shouldnt if (sof_core_debug & mask) suffice?

Reply:
Currently it is true, but while here I thought why not do it in a way that might be usable for the future, if ever someone needs to check for two set bits at the same time.

I'll leave it like this for now, but if more votes for the simple single bit check then I can change to that.

Reply:
@ujfalusi but the problem is if we do have more than 1 flag set in sof_core_debug, you need to make sure to check all the flags for this to be correct no?